src/lib/hooks/useStoredProfile.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useRef, useState } from 'react'
+import { isEqual } from 'lodash'
+
+import GDStore, { useCurriedSetters } from '../undux/GDStore'
+import userStorage from '../gundb/UserStorage'
+
+export default (withLocalState = false) => {
+  const store = GDStore.useStore()
+  const storedProfile = store.get('privateProfile')
+  const [setPrivateProfile] = useCurriedSetters(['privateProfile'])
+  const [profile, setProfile] = useState(() => (withLocalState ? storedProfile : null))
+
+  const initialProfileRef = useRef(storedProfile)
+  const initialLocalStateFlagRef = useRef(withLocalState)
+
+  useEffect(() => {
+    if (!isEqual(initialProfileRef.current, {})) {
+      return
+    }
+
+    // initialize profile value for first time from storedProfile in userStorage
+    userStorage.getProfile().then(loadedProfile => {
+      setPrivateProfile(loadedProfile)
+
+      if (initialLocalStateFlagRef.current) {
+        setProfile(loadedProfile)
```

Comment:
1. profile == privateprofile, why not set them both? why do we need the flag
2. what's the difference between profile and privateprofile?
3. this component needs more documentation, what's storedprofile?privateprofile?profile?what's the purpose of localstate?

Reply:
profile is local state var to be used in the profile edit component
I tried to keep it in the component itself but it's not possible to make it updated once after load due to the circiular deps between hooks